azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/AbstractAzureMojo.java

```diff
@@ -272,74 +274,97 @@ public int getHttpProxyPort() {
         return httpProxyPort;
     }
 
-    public Azure getAzureClient() throws AzureAuthFailureException {
+    public Azure getAzureClient() throws AzureAuthFailureException, AzureExecutionException {
         if (azure == null) {
             if (this.authentication != null && (this.authentication.getFile() != null || StringUtils.isNotBlank(authentication.getServerId()))) {
                 // TODO: remove the old way of authentication
                 Log.warn("You are using an old way of authentication which will be deprecated in future versions, please change your configurations.");
                 azure = new AzureAuthHelperLegacy(this).getAzureClient();
             } else {
-                initAuth();
-                azure = getAzureClientByAuthType();
+                azure = createAzureClient();
             }
             if (azure == null) {
                 getTelemetryProxy().trackEvent(INIT_FAILURE);
                 throw new AzureAuthFailureException(AZURE_INIT_FAIL);
             }
             printCurrentSubscription(azure);
-            getTelemetryProxy().addDefaultProperty(AUTH_TYPE, authType);
+            getTelemetryProxy().addDefaultProperty(AUTH_TYPE, getAuthType());
```

Comment:
I guest the `authType` may also be used in some other places. do you know when to use `authType` and `getAuthType()`, can they be merged?